packages/firestore/test/integration/api/array_transforms.test.ts

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import * as firestore from '@firebase/firestore-types';
+
+import firebase from '../util/firebase_export';
+import { apiDescribe, withTestDoc, withTestDb } from '../util/helpers';
+import { EventsAccumulator } from '../util/events_accumulator';
+
+// TODO(array-features): Remove "as any"
+// tslint:disable-next-line:no-any variable-name Type alias can be capitalized.
+const FieldValue = firebase.firestore.FieldValue as any;
+
+/**
+ * Note: Transforms are tested pretty thoroughly in server_timestamp.test.ts
+ * (via set, update, transactions, nested in documents, multiple transforms
+ * together, etc.) and so these tests mostly focus on the array transform
+ * semantics.
+ */
+// TODO(array-features): Enable these tests once the feature is available in
+// prod. For now you can run these tests using:
+// yarn test:browser --integration --local
+apiDescribe.skip('Array Transforms:', persistence => {
+  // A document reference to read and write to.
+  let docRef: firestore.DocumentReference;
+
+  // Accumulator used to capture events during the test.
+  let accumulator: EventsAccumulator<firestore.DocumentSnapshot>;
+
+  // Listener registration for a listener maintained during the course of the
+  // test.
+  let listenerRegistration: () => void;
+
+  /** Waits for a latency compensated local snapshot. */
+  async function waitForLocalEvent(): Promise<firestore.DocumentSnapshot> {
```

Comment:
Can we move this to helpers? This is all rock solid code, but having it twice doesn't make it all that much better :)